### PR TITLE
Added backport fix for reposync

### LINKED ISSUE #2865: cobbler reposync fails with error /var/www/cobbler/repo_mirror/<reponame>/.origin/repodata: File exists: 'Cannot create output directory'
### Describe the bug
cobbler reposync runs successfully for first time but fails after consecutive runs. 

### Steps to reproduce

1. Create new repo
2. Run # cobbler reposync  < - Successful
3. Run again # cobbler reposync < - Fails with error  "/var/www/cobbler/repo_mirror/<reponame>/.origin/repodata: File exists: 'Cannot create output directory'"

### Expected behavior
cobbler reposync should complete successfully on every run.

### Cobbler version

````
cobbler-3.2.2-4.module_el8+13419+741870f6.noarch
cobbler-web-3.2.2-4.module_el8+13419+741870f6.noarch
# cobbler --version
Cobbler 3.2.2
  source: ?, ?
  build time: Tue Nov 23 03:07:49 2021
````

### Operating system
CentOS Stream release 8

### Cobbler log

````
2021-12-17T08:57:58 - INFO | reposync; [None]
2021-12-17T08:59:41 - INFO | REMOTE invalid token; user(???)
2021-12-17T08:59:47 - INFO | authenticate; ['cobbler', True]
2021-12-17T08:59:48 - INFO | authenticate; ['cobbler', True]
2021-12-17T08:59:48 - DEBUG | authorize; ['cobbler', 'modify_repo', <cobbler.items.repo.Repo object at 0x7f53fdfb30b8>, None, True]
2021-12-17T08:59:48 - DEBUG | REMOTE cobbler authorization result: True; user(?)
2021-12-17T08:59:48 - INFO | REMOTE get_item(repo,CentOS-7-x86_64-base); user(?)
2021-12-17T08:59:48 - DEBUG | get_item; ['repo', 'CentOS-7-x86_64-base']
2021-12-17T08:59:48 - DEBUG | done with get_item; ['repo', 'CentOS-7-x86_64-base']
2021-12-17T08:59:48 - INFO | REMOTE get_settings; user(?)
2021-12-17T08:59:48 - DEBUG | REMOTE my settings are: {'allow_duplicate_hostnames': False, 'allow_duplicate_ips': False, 'allow_duplicate_macs': False, 'allow_dynamic_settings': False, 'always_write_dhcp_entries': False, 'anamon_enabled': False, 'auth_token_expiration': 3600, 'authn_pam_service': 'login', 'autoinstall_snippets_dir': '/var/lib/cobbler/snippets', 'autoinstall_templates_dir': '/var/lib/cobbler/templates', 'bind_chroot_path': '', 'bind_master': '127.0.0.1', 'boot_loader_conf_template_dir': '/etc/cobbler/boot_loader_conf', 'bootloaders_dir': '/var/lib/cobbler/loaders', 'grubconfig_dir': '/var/lib/cobbler/grub_config', 'build_reporting_enabled': False, 'build_reporting_email': ['root@localhost'], 'build_reporting_ignorelist': [], 'build_reporting_sender': '', 'build_reporting_smtp_server': 'localhost', 'build_reporting_subject': '', 'buildisodir': '/var/cache/cobbler/buildiso', 'cache_enabled': True, 'cheetah_import_whitelist': ['random', 're', 'time', 'netaddr'], 'client_use_https': False, 'client_use_localhost': False, 'cobbler_master': '', 'convert_server_to_ip': False, 'createrepo_flags': '-c cache -s sha', 'default_autoinstall': '/var/lib/cobbler/templates/default.ks', 'default_name_servers': [], 'default_name_servers_search': [], 'default_ownership': ['admin'], 'default_password_crypted': '$1$mF86/UHC$WvcIcX2t6crBz2onWxyac.', 'default_template_type': 'cheetah', 'default_virt_bridge': 'xenbr0', 'default_virt_disk_driver': 'raw', 'default_virt_file_size': 5, 'default_virt_ram': 512, 'default_virt_type': 'xenpv', 'enable_gpxe': False, 'enable_menu': True, 'http_port': 80, 'include': ['/etc/cobbler/settings.d/*.settings'], 'iso_template_dir': '/etc/cobbler/iso', 'jinja2_includedir': '/var/lib/cobbler/jinja2', 'kernel_options': {}, 'ldap_anonymous_bind': True, 'ldap_base_dn': 'DC=example,DC=com', 'ldap_port': 389, 'ldap_search_bind_dn': '', 'ldap_search_passwd': '', 'ldap_search_prefix': 'uid=', 'ldap_server': 'ldap.example.com', 'ldap_tls': True, 'ldap_tls_cacertfile': '', 'ldap_tls_certfile': '', 'ldap_tls_keyfile': '', 'bind_manage_ipmi': True, 'manage_dhcp': False, 'manage_dns': False, 'manage_forward_zones': [], 'manage_reverse_zones': [], 'manage_genders': False, 'manage_rsync': False, 'manage_tftpd': True, 'mgmt_classes': [], 'mgmt_parameters': {'from_cobbler': True}, 'next_server': '127.0.0.1', 'nsupdate_enabled': False, 'nsupdate_log': '/var/log/cobbler/nsupdate.log', 'nsupdate_tsig_algorithm': 'hmac-sha512', 'nsupdate_tsig_key': ['cobbler_update_key.', 'hvnK54HFJXFasHjzjEn09ASIkCOGYSnofRq4ejsiBHz3udVyGiuebFGAswSjKUxNuhmllPrkI0HRSSmM2qvZug=='], 'power_management_default_type': 'ipmilan', 'proxy_url_ext': '', 'proxy_url_int': '', 'puppet_auto_setup': False, 'puppet_parameterized_classes': True, 'puppet_server': '', 'puppet_version': 2, 'puppetca_path': '/usr/bin/puppet', 'pxe_just_once': True, 'nopxe_with_triggers': True, 'redhat_management_permissive': False, 'redhat_management_server': 'xmlrpc.rhn.redhat.com', 'redhat_management_key': '', 'register_new_installs': False, 'remove_old_puppet_certs_automatically': False, 'replicate_repo_rsync_options': '-avzH', 'replicate_rsync_options': '-avzH', 'reposync_flags': '-n --delete', 'reposync_rsync_flags': '-rltDv --copy-unsafe-links', 'restart_dhcp': True, 'restart_dns': True, 'run_install_triggers': True, 'scm_track_enabled': False, 'scm_track_mode': 'git', 'scm_track_author': 'cobbler <cobbler@localhost>', 'scm_push_script': '/bin/true', 'serializer_pretty_json': False, 'server': '172.16.200.168', 'sign_puppet_certs_automatically': False, 'signature_path': '/var/lib/cobbler/distro_signatures.json', 'signature_url': 'https://cobbler.github.io/signatures/3.0.x/latest.json', 'tftpboot_location': '/var/lib/tftpboot', 'virt_auto_boot': True, 'webdir': '/var/www/cobbler', 'webdir_whitelist': ['misc', 'web', 'webui', 'localmirror', 'repo_mirror', 'distro_mirror', 'images', 'links', 'pub', 'repo_profile', 'repo_system', 'svc', 'rendered', '.link_cache'], 'xmlrpc_port': 25151, 'yum_distro_priority': 1, 'yum_post_install_mirror': True, 'yumdownloader_flags': '--resolve'}; user(?)
2021-12-17T08:59:48 - INFO | REMOTE get_autoinstall_templates; user(?)
2021-12-17T08:59:48 - INFO | REMOTE version; user(cobbler)
2021-12-17T09:14:25 - INFO | REMOTE version; user(?)

````

## First Run - Successful 
````
running python triggers from /var/lib/cobbler/triggers/task/reposync/pre/*
running shell triggers from /var/lib/cobbler/triggers/task/reposync/pre/*
shell triggers finished successfully
hello, reposync
run, reposync, run!
creating: /var/www/cobbler/repo_mirror/CentOS-7-x86_64-base/config.repo
creating: /var/www/cobbler/repo_mirror/CentOS-7-x86_64-base/.origin/CentOS-7-x86_64-base.repo
running: /usr/bin/dnf reposync -n --delete --config=/var/www/cobbler/repo_mirror/CentOS-7-x86_64-base/.origin/CentOS-7-x86_64-base.repo --repoid=CentOS-7-x86_64-base -p /var/www/cobbler/repo_mirror -a x86_64
received on stdout: CentOS-7-x86_64-base                             39 MB/s |  25 MB     00:00
Last metadata expiration check: 0:00:09 ago on Fri 17 Dec 2021 08:49:27 AM UTC.
(1/4774): ElectricFence-2.2.2-39.el7.x86_64.rpm 2.3 MB/s |  35 kB     00:00
(2/4774): PyGreSQL-4.0-9.el7.x86_64.rpm         3.1 MB/s |  74 kB     00:00
(4770/4774): zsh-html-5.0.2-34.el7_8.2.x86_64.r 5.8 MB/s | 465 kB     00:00
(4771/4774): vsftpd-sysvinit-3.0.2-28.el7.x86_6 395 kB/s |  24 kB     00:00
(4772/4774): zsh-5.0.2-34.el7_8.2.x86_64.rpm     12 MB/s | 2.4 MB     00:00
(4773/4774): vte291-0.52.4-1.el7.x86_64.rpm     2.9 MB/s | 250 kB     00:00
(4774/4774): xerces-c-3.1.1-10.el7_7.x86_64.rpm 6.5 MB/s | 879 kB     00:00

received on stderr:
running: createrepo -g /var/www/cobbler/repo_mirror/CentOS-7-x86_64-base/.origin/repodata/2bdd2671151dd556f3dbfdf62d392c5a1f323d01-c7-x86_64-comps.xml -c cache -s sha /var/www/cobbler/repo_mirror/CentOS-7-x86_64-base
received on stdout: Directory walk started
Directory walk done - 4774 packages
Using 2bdd2671151dd556f3dbfdf62d392c5a1f323d01-c7-x86_64-comps.xml from target repo
Temporary output repo path: /var/www/cobbler/repo_mirror/CentOS-7-x86_64-base/.repodata/
Preparing sqlite DBs
Pool started (with 5 workers)
Pool finished

received on stderr:
running: chown -R root:apache /var/www/cobbler/repo_mirror/CentOS-7-x86_64-base
received on stdout:
received on stderr:
running: chmod -R 755 /var/www/cobbler/repo_mirror/CentOS-7-x86_64-base
received on stdout:
received on stderr:
### TASK COMPLETE ###
running python triggers from /var/lib/cobbler/triggers/task/reposync/post/*
running shell triggers from /var/lib/cobbler/triggers/task/reposync/post/*
shell triggers finished successfully

````

## Second Run - Failed
````
# cat 2021-12-17_085757_reposync.log
running python triggers from /var/lib/cobbler/triggers/task/reposync/pre/*
running shell triggers from /var/lib/cobbler/triggers/task/reposync/pre/*
shell triggers finished successfully
hello, reposync
run, reposync, run!
creating: /var/www/cobbler/repo_mirror/CentOS-7-x86_64-base/config.repo
creating: /var/www/cobbler/repo_mirror/CentOS-7-x86_64-base/.origin/CentOS-7-x86_64-base.repo
running: /usr/bin/dnf reposync -n --delete --config=/var/www/cobbler/repo_mirror/CentOS-7-x86_64-base/.origin/CentOS-7-x86_64-base.repo --repoid=CentOS-7-x86_64-base -p /var/www/cobbler/repo_mirror -a x86_64
received on stdout: CentOS-7-x86_64-base                             57 kB/s | 3.1 kB     00:00
[SKIPPED] ORBit2-2.14.19-13.el7.x86_64.rpm: Already downloaded
[SKIPPED] PyGreSQL-4.0-9.el7.x86_64.rpm: Already downloaded
[SKIPPED] ElectricFence-2.2.2-39.el7.x86_64.rpm: Already downloaded
[SKIPPED] PyPAM-0.5.0-19.el7.x86_64.rpm: Already downloaded
[SKIPPED] GConf2-3.2.6-8.el7.x86_64.rpm: Already downloaded
[SKIPPED] GConf2-devel-3.2.6-8.el7.x86_64.rpm: Already downloaded
...
..
[SKIPPED] xerces-c-3.1.1-10.el7_7.x86_64.rpm: Already downloaded

received on stderr:
Exception occurred: <class 'OSError'>
Exception value: [Errno unknown] Cannot create directory: /var/www/cobbler/repo_mirror/CentOS-7-x86_64-base/.origin/repodata: File exists: 'Cannot create output directory'
Exception Info:
  File "/usr/lib/python3.6/site-packages/cobbler/actions/reposync.py", line 161, in run
    self.sync(repo)

  File "/usr/lib/python3.6/site-packages/cobbler/actions/reposync.py", line 204, in sync
    self.yum_sync(repo)

  File "/usr/lib/python3.6/site-packages/cobbler/actions/reposync.py", line 622, in yum_sync
    h.perform(r)

  File "/usr/lib64/python3.6/site-packages/librepo/__init__.py", line 1595, in perform
    _librepo.Handle.perform(self, result)

reposync failed, tries left: 0
Exception occurred: <class 'cobbler.cexceptions.CX'>
Exception value: 'reposync failed, retry limit reached, aborting'
Exception Info:
  File "/usr/lib/python3.6/site-packages/cobbler/utils.py", line 95, in die
    raise CX(msg)

Exception occurred: <class 'cobbler.cexceptions.CX'>
Exception value: 'reposync failed, retry limit reached, aborting'
Exception Info:
  File "/usr/lib/python3.6/site-packages/cobbler/remote.py", line 98, in run
    rc = self._run(self)

  File "/usr/lib/python3.6/site-packages/cobbler/remote.py", line 326, in runner
    name=None, nofail=nofail, logger=self.logger)

  File "/usr/lib/python3.6/site-packages/cobbler/api.py", line 1375, in reposync
    action_reposync.run(name)

  File "/usr/lib/python3.6/site-packages/cobbler/actions/reposync.py", line 182, in run
    utils.die(self.logger, "reposync failed, retry limit reached, aborting")

  File "/usr/lib/python3.6/site-packages/cobbler/utils.py", line 103, in die
    raise CX(msg)

### TASK FAILED ###
````

### PR DESCRIPTION
**Pull request by c4t3l**
## Linked Items

Fixes #2865 for v3.2.2 which is the current supported stable branch in epel8.

 

## Description

This PR backports some of the changes presented in #2942. Namely, the librepo "Cannot create output directory" error and the missing noarch package downloads referenced [in this comment](https://github.com/cobbler/cobbler/pull/2942#issuecomment-1046951371)

## Behaviour changes

Old: Reposyncs fail on consecutive runs and noarch packages are ignored in repos.

New: Reposyncs pass on consecutive runs and noarch packages are downloaded.

## Category

This is related to a:

- [X] Bugfix
- [ ] Feature
- [ ] Packaging
- [ ] Docs
- [ ] Code Quality
- [ ] Refactoring
- [ ] Miscellaneous

## Tests

- [ ] Unit-Tests were created
- [ ] System-Tests were created
- [ ] Code is already covered by Unit-Tests
- [ ] Code is already covered by System-Tests
- [ ] No tests required

### CODE REVIEW

File: cobbler/actions/reposync.py
Comment (opoplawski) [inline cobbler/actions/reposync.py:579]: Personally it seems kind of odd to note the backport in a comment in the code.

Comment (c4t3l) [reply cobbler/actions/reposync.py:579]: I will go ahead and remove the comment.  It doesn't really provide any real benefit.

File: cobbler/actions/reposync.py
Comment (SchoolGuy) [inline cobbler/actions/reposync.py:576]:
```suggestion
        repodata_path = os.path.join(dest_path, "repodata")
```

See https://github.com/cobbler/cobbler/pull/2942/files#diff-77204e56ea1a781e3396e07cc7dea840201908389aa0a4bb0e00a6752d2a1e68R512

File: cobbler/actions/reposync.py
Comment (SchoolGuy) [inline cobbler/actions/reposync.py:448]: We did not have enums but I am not sure if this is the correct. I would believe that this is more fitting for the following (which would mean we include the arch):

https://github.com/cobbler/cobbler/pull/2942/files#diff-77204e56ea1a781e3396e07cc7dea840201908389aa0a4bb0e00a6752d2a1e68R450-R451

File: cobbler/actions/reposync.py
Comment (SchoolGuy) [inline cobbler/actions/reposync.py:583]:
```suggestion
            self.logger.info("Deleted old repo metadata for %s", repodata_path)
```

This is a logger and thus no percent formatting. See here: https://docs.python.org/3/howto/logging.html#logging-variable-data

**Review by SchoolGuy — COMMENTED** [review]
Many thanks for using the issue template. That makes this a lot easier. I have some comments that I would love if we address them before we merge this PR.

File: cobbler/actions/reposync.py
Comment (c4t3l) [reply cobbler/actions/reposync.py]: I made the requested changes.

File: cobbler/actions/reposync.py
Comment (c4t3l) [reply cobbler/actions/reposync.py]: I made the requested changes.

File: cobbler/actions/reposync.py
Comment (c4t3l) [reply cobbler/actions/reposync.py:448]: I'm still working through this component.  I am uncertain of the best approach.  According to `DNF-REPOSYNC(8)`:
```
       -a <architecture>, --arch=<architecture>
              Download only packages of given architectures (default is all architectures). Can be used multiple times.
```

Would simply adding `cmd = "%s -a %s -a noarch" % (cmd, repo.arch)` here be sufficient?

File: cobbler/actions/reposync.py
Comment (c4t3l) [reply cobbler/actions/reposync.py]: Ok... This is actually failing for me now:
```
received on stderr: 
Exception occurred: <class 'TypeError'>
Exception value: info() takes 2 positional arguments but 3 were given
Exception Info:
  File "/usr/lib/python3.6/site-packages/cobbler/actions/reposync.py", line 161, in run
    self.sync(repo)

  File "/usr/lib/python3.6/site-packages/cobbler/actions/reposync.py", line 204, in sync
    self.yum_sync(repo)

  File "/usr/lib/python3.6/site-packages/cobbler/actions/reposync.py", line 580, in yum_sync
    self.logger.info("Deleted old repo metadata for %s", repodata_path)
```

I'm gonna work through and resubmit a fix for this.

Comment (SchoolGuy) [reply cobbler/actions/reposync.py]: Oh damn. That could be my error. Is 3.2.x still using the custom `clogger.py` implementation? If yes then revert the applied change since my change will only work if the default Python logging module is used.

File: cobbler/actions/reposync.py
Comment (SchoolGuy) [reply cobbler/actions/reposync.py:448]: I have inherited this code. I was not yet able to work my way through to this line. I am dependant on your call here. I don't have a test Cobbler at hand where I can experiment with such things sadly.

Comment (c4t3l) [reply cobbler/actions/reposync.py:448]: In my local testing, this works for all arches.

File: cobbler/actions/reposync.py
Comment (c4t3l) [reply cobbler/actions/reposync.py]: Yes, sadly, 3.2.x is still using the `clogger.py` implementation.  I will revert this change and push.

File: cobbler/actions/reposync.py
Comment (c4t3l) [reply cobbler/actions/reposync.py]: Running more local introspective testing this morning I'm getting this:
```
received on stderr: 
Deleted old repo metadata for /var/www/cobbler/repo_mirror/eog-local8-testing/.origin/repodata
Exception occurred: <class 'OSError'>
Exception value: [Errno unknown] Cannot open /var/www/cobbler/repo_mirror/eog-local8-testing/.origin/repodata/repomd.xml: No such file or directory: 'Input/Output error'
Exception Info:
  File "/usr/lib/python3.6/site-packages/cobbler/actions/reposync.py", line 161, in run
    self.sync(repo)

  File "/usr/lib/python3.6/site-packages/cobbler/actions/reposync.py", line 204, in sync
    self.yum_sync(repo)

  File "/usr/lib/python3.6/site-packages/cobbler/actions/reposync.py", line 597, in yum_sync
    h.perform(r)

  File "/usr/lib64/python3.6/site-packages/librepo/__init__.py", line 1595, in perform
    _librepo.Handle.perform(self, result)
```
Reverting this change results in successful task completion:
```
[root@default-centos8-qemu ~]# cobbler reposync --only=eog-local8-testing
task started: 2022-08-18_101136_reposync
task started (id=Reposync, time=Thu Aug 18 10:11:36 2022)
running python triggers from /var/lib/cobbler/triggers/task/reposync/pre/*
running shell triggers from /var/lib/cobbler/triggers/task/reposync/pre/*
shell triggers finished successfully
hello, reposync
run, reposync, run!
creating: /var/www/cobbler/repo_mirror/eog-local8-testing/config.repo
creating: /var/www/cobbler/repo_mirror/eog-local8-testing/.origin/eog-local8-testing.repo
running: /usr/bin/dnf reposync --newest-only --delete --remote-time --config=/var/www/cobbler/repo_mirror/eog-local8-testing/.origin/eog-local8-testing.repo --repoid=eog-local8-testing -p /var/www/cobbler/repo_mirror -a x86_64 -a noarch
3.0 kB     00:00    
[SKIPPED] actions-runner-2.285.1-3.eog.el8.x86_64.rpm: Already downloaded      
[SKIPPED] eog-devops-certs-3.0-1.eog.el8.noarch.rpm: Already downloaded        
[SKIPPED] epkg-minimal-1.2.0-1.el8.noarch.rpm: Already downloaded              
[SKIPPED] github-backup-utils-3.4.0-3.eog.el8.noarch.rpm: Already downloaded   
[SKIPPED] hypr-radius-7.2.0-1.eog.el8.noarch.rpm: Already downloaded           
[SKIPPED] insight-platform-collector-2.0.2-2.eog.el8.noarch.rpm: Already downloaded
[SKIPPED] koji-eog-builder-plugins-3.4-1.eog.el8.noarch.rpm: Already downloaded
[SKIPPED] koji-eog-hub-plugins-3.4-1.eog.el8.noarch.rpm: Already downloaded    
[SKIPPED] koji-hub-plugins-filesystem-1.1-1.eog.el8.noarch.rpm: Already downloaded
[SKIPPED] koji-hub-plugins-reposync-1.0-1.eog.el8.noarch.rpm: Already downloaded
[SKIPPED] koji-hub-plugins-tag2distrepo-2-1.eog.el8.noarch.rpm: Already downloaded
[SKIPPED] painless-password-rotation-0^20200128git8387611-2.eog.el8.noarch.rpm: Already downloaded
[SKIPPED] painless-password-rotation-template-0^20200128git8387611-2.eog.el8.noarch.rpm: Already downloaded
[SKIPPED] phpipam-httpd-conf-1.5.0-6.eog.el8.noarch.rpm: Already downloaded    
[SKIPPED] phpipam-langpack-i18n-1.5.0-6.eog.el8.noarch.rpm: Already downloaded 
[SKIPPED] phpipam-server-1.5.0-6.eog.el8.noarch.rpm: Already downloaded        
[SKIPPED] rabbitmq-message-deduplication-0.4.4-4.eog.el8.noarch.rpm: Already downloaded
[SKIPPED] saltdk-2021.9.3-3.eog.el8.noarch.rpm: Already downloaded             
[SKIPPED] tinystatus-0^20210709gitfc128ad-5.eog.el8.noarch.rpm: Already downloaded
[SKIPPED] tinystatus-httpd-conf-0^20210709gitfc128ad-5.eog.el8.noarch.rpm: Already downloaded

received on stderr: 
running: createrepo  -c cache -s sha /var/www/cobbler/repo_mirror/eog-local8-testing
received on stdout: Directory walk started
Directory walk done - 20 packages
Temporary output repo path: /var/www/cobbler/repo_mirror/eog-local8-testing/.repodata/
Preparing sqlite DBs
Pool started (with 5 workers)
Pool finished

received on stderr: 
running: chown -R root:apache /var/www/cobbler/repo_mirror/eog-local8-testing
received on stdout: 
received on stderr: 
running: chmod -R 755 /var/www/cobbler/repo_mirror/eog-local8-testing
received on stdout: 
received on stderr: 
*** TASK COMPLETE ***
```

I will push my changes back into this PR.

File: cobbler/actions/reposync.py
Comment (SchoolGuy) [reply cobbler/actions/reposync.py:448]: Then let's resolve this conversation and let it stay as-is.

File: cobbler/actions/reposync.py
Comment (SchoolGuy) [reply cobbler/actions/reposync.py:576]: @c4t3l Is it possible that you check if this is then a problem on our development version? Because if this works on `main`, I am confused why it doesn't work here...

Comment (c4t3l) [reply cobbler/actions/reposync.py:576]: I checked out main (9d35aa29ab3ee19598dc16d9c1e8c70eac947d44) and built the rpm package (for epel8 in my case).  Please see output below:
```
[root@default-centos8-qemu ~]# cobbler reposync --only=eog-local8-testing
task started: 2022-08-18_154407_reposync
task started (id=Reposync, time=Thu Aug 18 15:44:07 2022)
reposync; ['eog-local8-testing']
hello, reposync
run, reposync, run!
running: /usr/bin/dnf reposync --newest-only --delete --refresh --remote-time --config=/var/www/cobbler/repo_mirror/eog-local8-testing/.origin/eog-local8-testing.repo --repoid=eog-local8-testing -p /var/www/cobbler/repo_mirror -a x86_64
190 kB     00:00    
running: createrepo  -c cache -s sha /var/www/cobbler/repo_mirror/eog-local8-testing
received on stdout: Directory walk started
running: chown -R root:apache /var/www/cobbler/repo_mirror/eog-local8-testing
received on stdout: 
running: chmod -R 755 /var/www/cobbler/repo_mirror/eog-local8-testing
received on stdout: 
*** TASK COMPLETE ***
```

Cobbler log output:
```
[Thread-4] 2022-08-18T15:44:07 - INFO | REMOTE start_task(Reposync); event_id(2022-08-18_154407_reposync); user(?)
[2022-08-18_154407_reposync] 2022-08-18T15:44:08 - INFO | reposync; ['eog-local8-testing']
[2022-08-18_154407_reposync] 2022-08-18T15:44:08 - INFO | hello, reposync
[2022-08-18_154407_reposync] 2022-08-18T15:44:08 - INFO | run, reposync, run!
[2022-08-18_154407_reposync] 2022-08-18T15:44:08 - INFO | running: /usr/bin/dnf reposync --newest-only --delete --refresh --remote-time --config=/var/www/cobbler/repo_mirror/eog-local8-testing/.origin/eog-local8-testing.repo --repoid=eog-local8-testing -p /var/www/cobbler/repo_mirror -a x86_64
[2022-08-18_154407_reposync] 2022-08-18T15:44:14 - INFO | received on stdout: eog-local8-testing                              785 kB/s | 190 kB     00:00    
actions-runner-2.285.1-3.eog.el8.x86_64.rpm      49 MB/s | 104 MB     00:02    

[2022-08-18_154407_reposync] 2022-08-18T15:44:14 - INFO | running: createrepo  -c cache -s sha /var/www/cobbler/repo_mirror/eog-local8-testing
[2022-08-18_154407_reposync] 2022-08-18T15:44:16 - INFO | received on stdout: Directory walk started
Directory walk done - 1 packages
Temporary output repo path: /var/www/cobbler/repo_mirror/eog-local8-testing/.repodata/
Preparing sqlite DBs
Pool started (with 5 workers)
Pool finished

[2022-08-18_154407_reposync] 2022-08-18T15:44:16 - INFO | running: chown -R root:apache /var/www/cobbler/repo_mirror/eog-local8-testing
[2022-08-18_154407_reposync] 2022-08-18T15:44:16 - INFO | received on stdout: 
[2022-08-18_154407_reposync] 2022-08-18T15:44:16 - INFO | running: chmod -R 755 /var/www/cobbler/repo_mirror/eog-local8-testing
[2022-08-18_154407_reposync] 2022-08-18T15:44:16 - INFO | received on stdout: 
[2022-08-18_154407_reposync] 2022-08-18T15:44:16 - INFO | ### TASK COMPLETE ###
```

So in this instance it worked, but only pulled down one x86_64 package.  I guess next step would be firing up a python debugger to see why the `dest_path` variable fails in v3.2.2?

Comment (SchoolGuy) [reply cobbler/actions/reposync.py:576]: Nah don't bother for the validation of this PR. This PR is now fine. If you desire to investigate then feel free to do so.

**Review by SchoolGuy — APPROVED** [review]
LGTM

**Comment by SchoolGuy** [comment]
@c4t3l Could you sign the commits please? (And if possible squash the fix and revert fix into the main commit)

**Comment by c4t3l** [comment]
@SchoolGuy, sure thing.  I will do that once I get back home.

**Comment by c4t3l** [comment]
Please let me know if there is anything else required.

**Comment by SchoolGuy** [comment]
@c4t3l Nope all was fine now. This PR just slipped under my radar.

**Pull request merged**
